### PR TITLE
 adds trip mvc with passing tests for show and index as well as links for prev, next 30. closes #6 closes #7

### LINKED ISSUE #6: Trips index
As a visitor,
When I visit the trips index,
I see the first 30 trips (duration, start date, start station, end date, end station, bike id, subscription type, zip code),
I also see a button to see more pages of trips,

When I visit a second page, there should be buttons to move both forward and backward in time.

** All Attributes must be present **

### LINKED ISSUE #7: Trips Show
As a visitor,
When I visit the trip show,
I see all attributes for that trip.

### PR DESCRIPTION
**Pull request by slimecog**
Description of Changes:
-
* create trips mvc
* create basic trips index page with offset links for previous, next 30 trips
*create basic trip show page

### Tests Added:
*trip model test
* visitor trips index test
* visitor trip show test

#### Reference Related Issue(s) or Spec Requirement:
-#. - 6, 7

#### Checklist:
- [x] Code follows code style of this project
- [x] Tests added to cover changes
- [x] All new and existing tests passing
#

#### Questions / Additional Notes:

@mgmilton @slimecog @annaroyer

### CODE REVIEW

File: app/controllers/trips_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class TripsController < ApplicationController
+
+  def index
+    @trips = Trip.order(:end_date).reverse_order.offset(params[:offset]).first(30)
```
Comment (mgmilton) [inline app/controllers/trips_controller.rb:4]: This params[:offset] is a really cool technique.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :trips
```
Comment (mgmilton) [inline config/routes.rb:3]: I would limit the routes to just the ones the controller has methods for at this point.

File: spec/features/trips/user_sees_trip_index_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        expect(page).to have_content(trip.end_station_id)
+        expect(page).to have_content(trip.bike_id)
+        expect(page).to have_content(trip.subscription_type)
+        expect(page).to have_content(trip.zip_code)
+      end
+      expect(page).to have_content("30 Most Recent Trips")
+      expect(page).to_not have_content(trip_31.bike_id)
+    end
+
+    it "shows a link to a page with more trips" do
```
Comment (mgmilton) [inline spec/features/trips/user_sees_trip_index_page_spec.rb:25]: This is a beastly test.

**Review by mgmilton — APPROVED** [review]
Great work Cam! Really into the offset params method.

**Comment by slimecog** [comment]
just kidding, rebased first

**Pull request closed without merging**